File: 94-binary-tree-inorder-traversal/binary-tree-inorder-traversal.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public List<Integer> inorderTraversal(TreeNode root) {
        Stack<TreeNode> q=new Stack<>();
        List<Integer> ans=new ArrayList<>();
        TreeNode node=root,ptr=null;

        if(root==null)
          return ans;
        q.push(root);
        while(!q.isEmpty()){
            while(node.left!=null){
                node=node.left;
                q.push(node);
            }
            while(!q.isEmpty()){
                node=q.pop();
                ans.add(node.val);
                if(node.right!=null){
                   node=node.right;
                   q.push(node);
                   break;
              }  
            }
               
        }
        return ans;
    }
}

File: 94-binary-tree-inorder-traversal/binary-tree-inorder-traversal.py
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution(object):
    def inorderTraversal(self, root):
        """
        :type root: TreeNode
        :rtype: List[int]
        """
        stack = []
        ans = []
        ptr = root

        while True:
            while ptr != None:
                stack.append(ptr)
                ptr = ptr.left
            
            if len(stack) != 0:
                ptr = stack[-1]
                stack.pop(-1)
                ans.append(ptr.val)
                ptr = ptr.right
            else:
                break
            
        return ans



        